src/vmspawn/vmspawn-util.c

```diff
@@ -270,6 +316,10 @@ int find_ovmf_config(int search_sb, OvmfConfig **ret) {
 
         assert(ret);
 
+        const char *native_arch_qemu = architecture_to_qemu_table[native_architecture()];
+        if (!native_arch_qemu)
+                return log_debug_errno(SYNTHETIC_ERRNO(EOPNOTSUPP), "Architecture %s not supported by qemu", architecture_to_string(native_architecture()));
```

Comment:
Maybe, better to split out this check, and use both in `find_ovmf_config()` and `find_qemu_binary()`.